src/Controller/AbsenceController.php

```diff
@@ -15,59 +16,271 @@ class AbsenceController extends BaseController
+        $sites = null;
+        if ($nbSites>1) {
+            $sites = array();
+            for ($i = 1; $i < 11; $i++) {
+                if (in_array((200 + $i), $droits)) {
+                    $sites[] = $i;
                 }
+            }
+        }
 
-                $filename = $file->getClientOriginalName();
+        $a = new \absences();
+        $a->groupe = true;
+        if ($agents_supprimes) {
+            $a->agents_supprimes = array(0,1);
+        }
+        $a->fetch(null, $perso_id, $debutSQL, $finSQL, $sites);
 
-                $ad = new AbsenceDocument();
-                $ad->absence_id($result['id']);
-                $ad->filename($filename);
-                $ad->date(new \DateTime());
-                $this->entityManager->persist($ad);
-                $this->entityManager->flush();
+        $absences = $a->elements;
 
-                $file->move(__DIR__ . AbsenceDocument::UPLOAD_DIR . $result['id'] . '/' . $ad->id(), $filename);
+        $adminOnly = $this->config('Absences-adminSeulement');
 
+        $sort = false;
+        // Tri par défaut du tableau
+        if ($admin or (!$adminOnly and in_array(6, $droits))) {
+            $sort = true;
+        }
+
+        $selectedAgents = array();
+        if ($admin) {
+            $selected = $perso_id == 0 ? "selected='selected'" : null;
+            $p = new \personnel();
+            if ($agents_supprimes) {
+                $p->supprime = array(0,1);
+            }
+            $p->responsablesParAgent = true;
+            $p->fetch();
+            $agents_menu = $p->elements;
+
+            // Filtre pour n'afficher que les agents gérés si l'option "Absences-notifications-agent-par-agent" est cochée
+            if ($this->config('Absences-notifications-agent-par-agent') and !$adminN2) {
+                $tmp = array();
+                foreach ($agents_menu as $elem) {
+                    if ($elem['id'] > 2) {
+                        if ($elem['id'] == $_SESSION['login_id']) {
+                            $tmp[$elem['id']] = $elem;
+                        } else {
+                            foreach ($elem['responsables'] as $resp) {
+                                if ($resp['responsable'] == $_SESSION['login_id']) {
+                                    $tmp[$elem['id']] = $elem;
+                                    break;
+                                }
+                            }
+                        }
+                    } else {
+                        continue;
+                    }
+                }
+                $agents_menu = $tmp;
             }
 
-            $msg = $result['msg'];
-            $msg2 = $result['msg2'];
-            $msg2_type = $result['msg2_type'];
+            // Liste des agents à conserver :
+            $perso_ids = array_keys($agents_menu);
+            foreach ($agents_menu as $agent) {
+                $selected = $agent['id'] == $perso_id ? "selected='selected'":null;
```

Comment:
selected or not selected should be done in the template

Reply:
The display is managed in the template. This variable is only meant to be used as a test to add the agent to the selectedAgents array

Reply:
To raise the ambiguity, the variable will be set to true instead of css properties